tutorials/three-js-tool-tutorial.md

```diff
@@ -0,0 +1,371 @@
+[three.js tutorial](https://threejs.org/docs/index.html#manual/en/introduction/Creating-a-scene).
+
+### Step 5.
+
+```javascript
+// 5. Initialize the three.js scene with a camera and contents
+camera = new THREE.PerspectiveCamera(70, rendererWidth / rendererHeight, 1, 1000);
+scene = new THREE.Scene();
+containerObj = new THREE.Object3D();
+containerObj.matrixAutoUpdate = false;
+scene.add(containerObj);
+```
+
+In addition to the `renderer` defined in the previous step, all three.js projects will need a `scene` to contain all
+the 3D objects, and a `camera` to define what perspective to view the scene from. Here, we first create a camera
+with a few parameters - the field of view, aspect ratio, and the "near and far planes" (if something is too close
+or too far away from the camera, it won't get rendered). We then create a new (empty) scene and add an object to
+it called the `containerObj`. Any 3D content that you want to see should get added to this `containerObj`, as we
+will later use the `spatialInterface` APIs to reposition it based on the movement of the phone. The way that we'll
+set the position of the `containerObj` requires us to set `matrixAutoUpdate` to false.       
+
+### Step 6.
+
+```javascript
+// 6. Add some content to the scene's container object - you can customize this!
+let texture = new THREE.TextureLoader().load('textures/crate.gif');
+let geometry = new THREE.BoxBufferGeometry(100, 100, 100);
+let material = new THREE.MeshBasicMaterial({map: texture});
+mesh = new THREE.Mesh(geometry, material);
+mesh.position.setZ(50);
+containerObj.add(mesh);
+```
+
+Here's the real meat of the application. Everything else in this tutorial is pretty much the scaffolding that makes
+this scene run, but this step is where we put the 3D contents into the scene. To test it out, we add a box to the
+scene. You can [download the crate.gif](https://github.com/mrdoob/three.js/blob/master/examples/textures/crate.gif)
+file from the three.js examples and add it to a new directory in your tool called `textures`.
+
+Feel free to play around with this step and swap out the crate with other 3D content. Be aware that many of the other
+three.js examples will require you to include additional scripts - try the crate first to make sure it works.
+
+### Step 7.
+
+```javascript
+// 7. Use the Spatial Toolbox APIs to prepare to render correctly
+spatialInterface.setFullScreenOn();
+spatialInterface.addMatrixListener(renderScene);
+```
+
+Here are the two key lines where we use the Spatial Toolbox APIs to make this 3D scene work.
+
+`setFullScreenOn` makes this tool ignore the regular Spatial Toolbox rendering, which normally tries to render this
+tool as a 2D panel floating somewhere in 3D space. When we call this, the tool instead "sticks" to the screen and
+resizes to fill the full width and height of the screen. On it's own, this breaks the AR behavior of this tool.
```

Comment:
grammar nit: should be "On its own"

Reply:
ah good catch - thanks for the thorough review!